packages/block-library/src/site-title/edit.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp, useEntitySaving } from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { RichText } from '@wordpress/block-editor';
+
+export default function SiteTitleEdit() {
+	const [ title, setTitle ] = useEntityProp( 'root', 'site', 'title' );
+	const [ isDirty, isSaving, save ] = useEntitySaving( 'root', 'site', 'title' );
+	return (
+		<>
+			<Button
+				isPrimary
+				className="wp-block-site-title__save-button"
+				disabled={ ! isDirty || ! title }
+				isBusy={ isSaving }
+				onClick={ save }
+			>
+				{ __( 'Update' ) }
+			</Button>
```

Comment:
We can probably remove that button once we land the other PR (multi-entity saving)

Reply:
Yep!